// ///////////////////////////////////////////////////////////////////////////////////////
//  Copyright (c) 2018 Logan Barnes - All Rights Reserved
//  Geometry Visualisation Server
// ///////////////////////////////////////////////////////////////////////////////////////
syntax = "proto3";

package gvs.proto;

import "google/protobuf/empty.proto";
import "gvs/types.proto";

service Scene {
    rpc add_item (SceneItemInfo) returns (SceneResponse);

    rpc scene_updates (google.protobuf.Empty) returns (stream SceneUpdate);
}

message SceneResponse {
    string error_msg = 1; // empty if rpc was successful
}

message SceneUpdate {
    oneof update_type {
        SceneItemInfo added_item = 1;
        // SceneItemInfo updated_item = 2;
        // SceneItemInfo removed_item = 3;
    }
}
